docs/dev-guides/PERSONAL-ENVIRONMENTS.md

```diff
@@ -0,0 +1,523 @@
+[terraform-ap-depoyments](https://github.com/broadinstitute/terraform-ap-deployments)
+repo, and the subject is any cloud dependencies that must be instantiated for the new environment. Different
+use cases will have different needs here, but for Workspace Manager we're interested in the following.
+
+The Broad's Terraform modules live in a public repository called [terraform-ap-modules](https://github.com/broadinstitute/terraform-ap-modules).
+These are combined with generated private client modules, variables, and secrets to instantiate the
+actual environments and artifacts. The latter are contained in the private repo [terraform-ap-deployments](https://github.com/broadinstitute/terraform-ap-deployments), 
+and this is the one we need to change. See this [DevOps Portal entry](https://docs.dsp-devops.broadinstitute.org/mc-terra/mcterra-deployment#Infrastructure---Terraform)
+for more background.
+
+First, we declare a new Terra Environment in [atlantis.yaml](https://github.com/broadinstitute/terraform-ap-deployments/blob/master/atlantis.yaml#L70). A new object
+in a yaml array describes the top-level properties of the environment. Note that "environment" here
+is a set of conventions for creating a stand-alone instance of one or more services, and not anything
+specific in k8s or Terraform. Mine looks like this:
+```yaml
+- name: terra-env-jcarlton
+  dir: terra-env
+  workflow: terra-env-jcarlton
+  workspace: jcarlton
+  autoplan:
+    enabled: false
+```
+It's easy to get confused between the env name (`terra-env-jcarlton`), the workspace (`jcarlton`), and the k8s namespace eventually
+created (`terra-jcarlton`). There are numerous examples of the conventions in this file, and I admit
+I cheated a little off the surrounding entries. Note that Atlantis is a tool for managing Terraform updates,
+but atlantis.yaml is not itself a Terraform file.
+
+Second, we specify some workflow settings for deploying the environment. Among other things, this tells
+Atlantis to pass the variables file `jcarlton.tfvars`
+```yaml
+  terra-env-jcarlton:
+    plan:
+      steps:
+        - init:
+            extra_args: ["-upgrade"]
+        - plan:
+            extra_args: ["-var-file", "tfvars/jcarlton.tfvars"]
+```
+
+Third, we check in a tfvars file for Terraform to use as its input variables when building this [workspace](https://www.terraform.io/docs/state/workspaces.html).
+```hcl-terraform
+# General
+google_project = "terra-kernel-k8s"
+cluster        = "integration"
+cluster_short  = "integ"
+
+terra_apps = {
+  workspace_manager = true,
+  crl_janitor       = true,
+  buffer            = true,
+}
+
+# Resource Buffer Service Testing
+buffer_google_folder_ids   = ["637867149294"]
+buffer_billing_account_ids = ["01A82E-CA8A14-367457"]
+
+# Workspace Manager
+wsm_workspace_project_folder_id = "1094831421623" # DSP Integration
+wsm_billing_account_ids         = ["01A82E-CA8A14-367457"]
+```
```

Comment:
I would not paste this in here - I am about to change this and I am sure it will continue to evolve. Perhaps better to point at an existing personal file?

Reply:
Either way someone needs to update this when it changes. I prefer an end-to-end, self-contained-ish guide, but I'm certainly linking to plenty of other things.

I can indicate that the `tfvars` should line up with the variables specified in the modules, assuming those are easy to get to from here.